packages/shared/__tests__/define-vars/stylex-define-vars-test.js

```diff
@@ -186,6 +186,85 @@ describe('stylex-define-vars test', () => {
     `);
   });
 
+  test('converts set of vars with nested at rules to CSS and includes key in variable name as prefix in debug mode', () => {
+    const themeName = 'TestTheme.stylex.js//buttonTheme';
+    const classNamePrefix = 'x';
+    const defaultVars = {
+      bgColor: {
+        default: 'blue',
+        '@media (prefers-color-scheme: dark)': {
+          default: 'lightblue',
+          '@supports (color: oklab(0 0 0))': 'oklab(0.7 -0.3 -0.4)',
+        },
+        '@media print': 'white',
+      },
+      bgColorDisabled: {
+        default: {
+          default: 'grey',
+          '@supports (color: oklab(0 0 0))': 'oklab(0.7 -0.3 -0.4)',
+        },
+        '@media (prefers-color-scheme: dark)': {
+          default: 'rgba(0, 0, 0, 0.8)',
+          '@supports (color: oklab(0 0 0))': 'oklab(0.7 -0.3 -0.4)',
+        },
+      },
+      cornerRadius: '10px',
+      fgColor: {
+        default: 'pink',
+      },
+    };
+    const [jsOutput, cssOutput] = styleXDefineVars(defaultVars, {
+      themeName,
+      debug: true,
+    });
+
+    expect(jsOutput).toEqual({
+      __themeName__: classNamePrefix + createHash(themeName),
+      bgColor: `var(--${'bgColor-' + classNamePrefix + createHash(`${themeName}.bgColor`)})`,
+      bgColorDisabled: `var(--${
+        'bgColorDisabled-' +
+        classNamePrefix +
+        createHash(`${themeName}.bgColorDisabled`)
+      })`,
+      cornerRadius: `var(--${
+        'cornerRadius-' +
+        classNamePrefix +
+        createHash(`${themeName}.cornerRadius`)
+      })`,
+      fgColor: `var(--${'fgColor-' + classNamePrefix + createHash(`${themeName}.fgColor`)})`,
+    });
```

Comment:
Can be a bit simpler this way:

```suggestion
    expect(jsOutput).toEqual({
      __themeName__: classNamePrefix + createHash(themeName),
      bgColor: `var(--bgColor-${classNamePrefix + createHash(`${themeName}.bgColor`)})`,
      bgColorDisabled: `var(--bgColorDisabled-${
        classNamePrefix +
        createHash(`${themeName}.bgColorDisabled`)
      })`,
      cornerRadius: `var(--cornerRadius-${
        classNamePrefix +
        createHash(`${themeName}.cornerRadius`)
      })`,
      fgColor: `var(--fgColor-${classNamePrefix + createHash(`${themeName}.fgColor`)})`,
    });
```